src/platform/ESP32/ESP32DnssdImpl.cpp

```diff
@@ -418,8 +433,23 @@ static CHIP_ERROR ParseSrvResult(ResolveContext * ctx)
         ctx->mService->mSubTypeSize   = 0;
         if (ctx->mInterfaceId == chip::Inet::InterfaceId::Null())
         {
-            // If the InterfaceId in the context is Null, we will use the Station netif by default.
-            ctx->mService->mInterface = Inet::InterfaceId(DeviceLayer::Internal::ESP32Utils::GetStationNetif());
+            if (ctx->mSrvQueryResult->esp_netif)
+            {
+                ctx->mService->mInterface =
+                    Inet::InterfaceId(static_cast<struct netif *>(esp_netif_get_netif_impl(ctx->mSrvQueryResult->esp_netif)));
+            }
+            else
+            {
+                // If the InterfaceId in the context and esp_netif in current result is Null,
```

Comment:
Seems like this code should be factored out into a helper function, since we have it in two places that now have to stay in sync and the code is a bit complicated.

Reply:
Done